### PR TITLE
Add unit tests for `pg.ddl`

### PR DESCRIPTION
**Pull request by o-rumiantsev**
Need to fix getting domain from domains map, because there are no field `definition`.
https://github.com/metarhia/globalstorage/blob/007aba9a725cc8b1df5de020b4a6c1e3cf5a99a3/lib/pg.ddl.js#L518

### CODE REVIEW

**Comment by o-rumiantsev** [comment]
In Node.js version 11 Array.prototype.sort takes 1, 0 or -1 explictly, so we need to rewrite this line
https://github.com/metarhia/globalstorage/blob/007aba9a725cc8b1df5de020b4a6c1e3cf5a99a3/lib/pg.ddl.js#L301

**Comment by belochub** [comment]
@o-rumiantsev according to [ECMAScript specification](http://www.ecma-international.org/ecma-262/6.0/#sec-array.prototype.sort) `Array.prototype.sort` has always expected the `comparefn` to return negative value when `x < y`, zero when `x = y` and positive value when `x > y`, the problem with this comparison function is that it is not consistent, and according to the specification:
>If comparefn is not undefined and is not a consistent comparison function for the elements of this array (see below), the sort order is implementation-defined. The sort order is also implementation-defined if comparefn is undefined and SortCompare (22.1.3.24.1) does not act as a consistent comparison function.

Node.js 11 uses newer version of V8 engine where they have [changed](https://v8.dev/blog/array-sort) the `Array.prototype.sort` implementation, meaning that it also changes the way the arrays are being sorted when using a not consistent comparison function.

File: test/pg.ddl.unit.js [hunk 1]
```diff
@@ -0,0 +1,788 @@
...
+  unitTests.endAfterSubtests();
+
+  unitTests.testSync('pg.ddl pad', test =>
+    test.strictSame(ddl.pad('__NAME__', 18, '#'), '__NAME__##########')
+  );
+
+  unitTests.testSync('pg.ddl padProperty', test =>
+    test.strictSame(ddl.padProperty('__NAME__', 18), '"__NAME__"          ')
+  );
+
```
Comment (nechaido) [inline test/pg.ddl.unit.js]: You should also add tests, that will ensure function throws an error on invalid args: `padPeoperty('__Name__', 4)`.

File: test/pg.ddl.unit.js [hunk 2]
```diff
@@ -0,0 +1,788 @@
...
+          {
+            name: 'field',
+            property: { category: 'Schema4' },
+            destination: 'Schema4',
+          },
+        ],
+        ['Schema1', 'Schema2']
+      ),
+      [
+        [
```
Comment (nechaido) [inline test/pg.ddl.unit.js]: I'd be better to extract this arrays to variables named `unresolved` and `resolvableLinks` accordingly.

File: test/pg.ddl.unit.js [hunk 3]
```diff
@@ -0,0 +1,788 @@
...
+-- Category: LocalCategory1 ----------------------------------------------------
+
+CREATE TABLE "LocalCategory1" (
+  "Id"    bigserial,
+  "field" text
+);
+
+ALTER TABLE "LocalCategory1" ADD CONSTRAINT "pkLocalCategory1Id" \
+PRIMARY KEY ("Id");`,
+        unresolved: [],
```
Comment (nechaido) [inline test/pg.ddl.unit.js]: You should also add a test, to ensure that unresolved links are returned from this function.

File: test/pg.ddl.unit.js [hunk 4]
```diff
@@ -0,0 +1,1179 @@
...
+
+const { join } = require('path');
+const common = require('@metarhia/common');
+const metatests = require('metatests');
+const metaschema = require('metaschema');
+const ddl = require('../lib/pg.ddl');
+
+const schemasDir = join(__dirname, 'fixtures/ddl-unit');
+const unitTests = metatests.test('pg.ddl unit tests');
+metaschema.fs.loadAndCreate(schemasDir, { common }, (err, ms) => {
```
Comment (belochub) [inline test/pg.ddl.unit.js]: I'd prefer not to have 1168 lines long function here.

File: test/pg.ddl.unit.js [hunk 5]
```diff
@@ -0,0 +1,1179 @@
...
+    test.strictSame(
+      ddl.generateExtensions(['ext1', 'ext2', 'ext3']),
+      'CREATE EXTENSION IF NOT EXISTS ext1;\n' +
+      'CREATE EXTENSION IF NOT EXISTS ext2;\n' +
+      'CREATE EXTENSION IF NOT EXISTS ext3;\n'
+    )
+  );
+
+});
+
```
Comment (belochub) [inline test/pg.ddl.unit.js]: Redundant empty line.

**Comment by o-rumiantsev** [comment]
@belochub ping

File: test/unittests/pg.ddl/generateManyToMany.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  test.strictSame(
+    ddl.generateManyToMany(
+      'CategoryWithMany',
+      'LocalCategory1',
+      'field',
+      ms.categories,
+      'LocalCategory1'
+    ),
+    `
+CREATE TABLE "CategoryWithManyfield" (
```
Comment (nechaido) [inline test/unittests/pg.ddl/generateManyToMany.js]: If you use string concatenation it's better to not mix it with multiline template literals.

**Comment by o-rumiantsev** [comment]
@belochub this can be merged

**Comment by belochub** [comment]
Landed in https://github.com/metarhia/globalstorage/commit/6ea449050b4c0d27a73b7e20f837d3577fe09932 and https://github.com/metarhia/globalstorage/commit/1b6f3d5ddb0d7909528616516ae5deb68d01716c.

**Pull request closed without merging**